kiva/celiagg.py

```diff
@@ -826,6 +835,33 @@ def draw_path_at_points(self, points, path, mode=constants.FILL_STROKE):
             fill=self.fill_paint,
         )
 
+    def draw_marker_at_points(self, points, size,
+                              marker=constants.SQUARE_MARKER):
+        """ Draw a marker at a collection of points
+        """
+        # Apply the current transform
+        ctm = self.transform
+        self.marker_gc.transform(
+            ctm.sx, ctm.shy,
+            ctm.shx, ctm.sy,
+            ctm.tx, ctm.ty,
+        )
+
+        # Grab the fill and stroke colors (where possible)
+        fill = (0.0, 0.0, 0.0, 0.0)
+        stroke = (0.0, 0.0, 0.0, 1.0)
+        if isinstance(self.fill_paint, agg.SolidPaint):
+            fp = self.fill_paint
+            fill = (fp.r, fp.g, fp.b, fp.a)
+        if isinstance(self.stroke_paint, agg.SolidPaint):
+            sp = self.stroke_paint
+            stroke = (sp.r, sp.g, sp.b, sp.a)
```

Comment:
Maybe it would be better to add `set_stroke_color` and `set_fill_color` methods to `MarkerRenderer*` so that we can just pass through colors from `GraphicsContext.set_*_color` methods? This particular block of code is just here in case a gradient fill color was set.

Reply:
i'm not sure i fully understand what you're recommending here.

Reply:
Add `set_*_color` methods to `MarkerRenderer`, then in the `set_*_color` methods of `GraphicsContext` call `self.marker_gc.set_*_color(color)` so that the marker renderer stays in sync with the simple fill and stroke colors. If you set a gradient on the celiagg backend and then call `draw_marker_at_points`, the fill color of the markers will be transparent. One could argue that it's a bug in the code doing the drawing, but it's a silent failure.

Reply:
I won't doing anything about it in this PR